Intersect.Client/MonoGame/Graphics/MonoRenderer.cs

```diff
@@ -615,33 +610,43 @@ public override GameTileBuffer CreateTileBuffer()
             var origin = Vector2.Zero;
             if (Math.Abs(rotationDegrees) > 0.01)
             {
-                rotationDegrees = (float) (Math.PI / 180 * rotationDegrees);
+                rotationDegrees = (float)(Math.PI / 180 * rotationDegrees);
                 origin = new Vector2(sw / 2f, sh / 2f);
 
                 //TODO: Optimize in terms of memory AND performance.
                 var pnt = new Pointf(0, 0);
-                var pnt1 = new Pointf((float) tw, 0);
-                var pnt2 = new Pointf(0, (float) th);
-                var cntr = new Pointf((float) tw / 2, (float) th / 2);
+                var pnt1 = new Pointf(tw, 0);
+                var pnt2 = new Pointf(0, th);
+                var cntr = new Pointf(tw / 2, th / 2);
 
                 var pntMod = Rotate(pnt, cntr, rotationDegrees);
                 var pntMod2 = Rotate(pnt1, cntr, rotationDegrees);
                 var pntMod3 = Rotate(pnt2, cntr, rotationDegrees);
 
-                var width = (int) Math.Round(GetDistance(pntMod.X, pntMod.Y, pntMod2.X, pntMod2.Y));
-                var height = (int) Math.Round(GetDistance(pntMod.X, pntMod.Y, pntMod3.X, pntMod3.Y));
+                var width = (int)Math.Round(GetDistance(pntMod.X, pntMod.Y, pntMod2.X, pntMod2.Y));
+                var height = (int)Math.Round(GetDistance(pntMod.X, pntMod.Y, pntMod3.X, pntMod3.Y));
 
                 if (packRotated)
                 {
-                    var z = width;
-                    width = height;
-                    height = z;
+                    (width, height) = (height, width);
                 }
 
                 tx += width / 2f;
                 ty += height / 2f;
             }
 
+            // Cache the result of ConvertColor(renderColor) in a temporary variable.
+            var color = ConvertColor(renderColor);
+
+            // Use a single Draw method to avoid duplicating code.
+            void Draw(FloatRect currentView, GameRenderTexture targetObject)
+            {
+                StartSpritebatch(currentView, blendMode, shader, targetObject, false, null, drawImmediate);
+                mSpriteBatch.Draw((Texture2D)texture, new Vector2(tx, ty),
+                    new XNARectangle((int)sx, (int)sy, (int)sw, (int)sh), color, rotationDegrees, origin,
+                    new Vector2(tw / sw, th / sh), SpriteEffects.None, 0);
```

Comment:
Break each argument of Draw() into a separate line to make it more readable please, also the starting ( and terminating ); should be on separate lines from the first/last argument